app/controllers/assumed/update/HasInternationalTaxIdentifierController.scala

```diff
@@ -31,49 +30,45 @@ import java.time.Year
-                                              requireData: DataRequiredAction,
-                                              checkAssumedReportingAllowed: CheckAssumedReportingAllowedAction,
-                                              formProvider: HasInternationalTaxIdentifierFormProvider,
-                                              val controllerComponents: MessagesControllerComponents,
-                                              view: HasInternationalTaxIdentifierView
-                                            )(implicit ec: ExecutionContext)
+class HasInternationalTaxIdentifierController @Inject()(override val messagesApi: MessagesApi,
+                                                        sessionRepository: SessionRepository,
+                                                        identify: IdentifierAction,
+                                                        getData: DataRetrievalActionProvider,
+                                                        requireData: DataRequiredAction,
+                                                        checkAssumedReportingAllowed: CheckAssumedReportingAllowedAction,
+                                                        formProvider: HasInternationalTaxIdentifierFormProvider,
+                                                        val controllerComponents: MessagesControllerComponents,
+                                                        view: HasInternationalTaxIdentifierView)
+                                                       (using ExecutionContext)
   extends FrontendBaseController with I18nSupport with AnswerExtractor {
 
 
   def onPageLoad(operatorId: String, reportingPeriod: Year): Action[AnyContent] =
     (identify andThen checkAssumedReportingAllowed andThen getData(operatorId, Some(reportingPeriod)) andThen requireData) { implicit request =>
       getAnswers(AssumingOperatorNamePage, TaxResidencyCountryPage) { case (assumingOperatorName, country) =>
-  
-        val form = formProvider(assumingOperatorName, country)
-  
         val preparedForm = request.userAnswers.get(HasInternationalTaxIdentifierPage) match {
-          case None => form
-          case Some(value) => form.fill(value)
+          case None => formProvider(assumingOperatorName, country)
+          case Some(value) => formProvider(assumingOperatorName, country).fill(value)
         }
-  
+
         Ok(view(preparedForm, operatorId, reportingPeriod, assumingOperatorName, country))
       }
     }
 
   def onSubmit(operatorId: String, reportingPeriod: Year): Action[AnyContent] =
     (identify andThen checkAssumedReportingAllowed andThen getData(operatorId, Some(reportingPeriod)) andThen requireData).async { implicit request =>
       getAnswersAsync(AssumingOperatorNamePage, TaxResidencyCountryPage) { case (assumingOperatorName, country) =>
-  
+
         val form = formProvider(assumingOperatorName, country)
-  
+
         form.bindFromRequest().fold(
```

Comment:
Use formProvider(assumingOperatorName, country).bindFormRequest to be consistent with other changes above

Reply:
Done